tools/package-builder/src/core/schemas/entrypoint.ts

```diff
@@ -50,14 +50,20 @@ export const infoSchema = z
     asset: z.union([z.string(), artifacts.assetPackageSchema]).optional(),
     binary: softwareOptionsSchema.optional(),
     environment: environmentOptionsSchema.optional(),
+    docker: softwareOptionsSchema.optional(),
   })
   .refine(
-    (data) =>
-      toInt(data.reference) + toInt(data.asset) + toInt(data.binary) + toInt(data.environment) == 1,
+    (data) => {
+      const n = toInt(data.reference)
+        + toInt(data.asset)
+        + toInt(data.binary)
+        + toInt(data.environment);
+      return n === 1 || data.docker !== undefined;
```

Comment:
This allows strange things like `reference + docker` or `asset + docker`.
Not sure we should allow this.